ecs/context.go

```diff
@@ -20,100 +20,185 @@ import (
-	user prompt.UI
-}
-
-func newContextCreateHelper() contextCreateAWSHelper {
-	return contextCreateAWSHelper{
-		user: prompt.User{},
+func getEnvVars() ContextParams {
+	c := ContextParams{
+		Profile: os.Getenv("AWS_PROFILE"),
+		Region:  os.Getenv("AWS_REGION"),
+	}
+	if c.Region == "" {
+		defaultRegion := os.Getenv("AWS_DEFAULT_REGION")
+		if defaultRegion == "" {
+			defaultRegion = "us-east-1"
+		}
+		c.Region = defaultRegion
 	}
-}
 
-func (h contextCreateAWSHelper) createProfile(name string) error {
-	accessKey, secretKey, err := h.askCredentials()
+	p := credentials.EnvProvider{}
+	creds, err := p.Retrieve()
 	if err != nil {
-		return err
-	}
-	if accessKey != "" && secretKey != "" {
-		return h.saveCredentials(name, accessKey, secretKey)
+		return c
 	}
-	return nil
+	c.AccessKey = creds.AccessKeyID
+	c.SecretKey = creds.SecretAccessKey
+	return c
+}
+
+type contextCreateAWSHelper struct {
+	user             prompt.UI
+	availableRegions func(opts *ContextParams) ([]string, error)
 }
 
-func (h contextCreateAWSHelper) createContext(profile, region, description string) (interface{}, string) {
-	if profile == "default" {
-		profile = ""
+func newContextCreateHelper() contextCreateAWSHelper {
+	return contextCreateAWSHelper{
+		user:             prompt.User{},
+		availableRegions: listAvailableRegions,
 	}
-	description = strings.TrimSpace(
-		fmt.Sprintf("%s (%s)", description, region))
-	return store.EcsContext{
-		Profile: profile,
-		Region:  region,
-	}, description
 }
 
 func (h contextCreateAWSHelper) createContextData(_ context.Context, opts ContextParams) (interface{}, string, error) {
-	profile := opts.Profile
-	region := opts.Region
-
-	profilesList, err := h.getProfiles()
-	if err != nil {
-		return nil, "", err
-	}
-	if profile != "" {
-		// validate profile
-		if profile != "default" && !contains(profilesList, profile) {
-			return nil, "", errors.Wrapf(errdefs.ErrNotFound, "profile %q", profile)
+	if opts.CredsFromEnv {
+		// Explicit creation from ENV variables
+		ecsCtx, descr := h.createContext(&opts)
+		return ecsCtx, descr, nil
+	} else if opts.AccessKey != "" && opts.SecretKey != "" {
+		// Explicit creation using keys
+		err := h.createProfileFromCredentials(&opts)
+		if err != nil {
+			return nil, "", err
+		}
+	} else if opts.Profile != "" {
+		// Excplicit creation by selecting a profile
+		// check profile exists
+		profilesList, err := getProfiles()
+		if err != nil {
+			return nil, "", err
+		}
+		if !contains(profilesList, opts.Profile) {
+			return nil, "", fmt.Errorf("profile %q not found", opts.Profile)
 		}
 	} else {
-		// choose profile
-		profile, err = h.chooseProfile(profilesList)
+		// interactive
+		options := []string{
+			"AWS secret and token credentials",
+			"AWS environment variables",
+		}
+
+		if _, err := os.Stat(getAWSConfigFile()); err == nil {
+			// User has .asw/config file, so we can offer to select one of his profiles
+			options = append(options, "An existing AWS profile")
```

Comment:
If they have existing profiles, maybe this should be the first option? WDYT @ndeloof ?